aptos/contracts/sources/pyth.move

```diff
@@ -0,0 +1,810 @@
+            governance_emitter_chain_id,
+            governance_emitter_address,
+            parse_data_sources(
+                data_sources_emitter_chain_ids,
+                data_sources_emitter_addresses,
+            ),
+            update_fee
+        )
+    }
+
+    fun init_internal(
+        signer_capability: account::SignerCapability,
+        chain_id: u64,
+        stale_price_threshold: u64,
+        governance_emitter_chain_id: u64,
+        governance_emitter_address: vector<u8>,
+        data_sources: vector<DataSource>,
+        update_fee: u64) {
+        let pyth = account::create_signer_with_capability(&signer_capability);
+        state::init(
+            &pyth,
+            chain_id,
+            stale_price_threshold,
+            update_fee,
+            data_source::new(
+                governance_emitter_chain_id,
+                external_address::from_bytes(governance_emitter_address)),
+            data_sources,
+            signer_capability
+        );
+        event::init(&pyth);
+        coin::register<AptosCoin>(&pyth);
+    }
+
+    fun parse_data_sources(
+        emitter_chain_ids: vector<u64>,
+        emitter_addresses: vector<vector<u8>>): vector<DataSource> {
+        let sources = vector::empty();
+        let i = 0;
+        while (i < vector::length(&emitter_chain_ids)) {
+            vector::push_back(&mut sources, data_source::new(
+                *vector::borrow(&emitter_chain_ids, i),
+                external_address::from_bytes(*vector::borrow(&emitter_addresses, i))
+            ));
+
+            i = i + 1;
+        };
+
+        sources
+    }
+
+    #[test_only]
+    /// Expose a public initialization function for use in tests
+    public fun init_test(
+        signer_capability: account::SignerCapability,
+        chain_id: u64,
+        stale_price_threshold: u64,
+        governance_emitter_chain_id: u64,
+        governance_emitter_address: vector<u8>,
+        data_sources: vector<DataSource>,
+        update_fee: u64,
+    ) {
+        init_internal(
+            signer_capability,
+            chain_id,
+            stale_price_threshold,
+            governance_emitter_chain_id,
+            governance_emitter_address,
+            data_sources,
+            update_fee
+        )
+    }
+
+// -----------------------------------------------------------------------------
+// Update the cached prices
+
+    /// Update the cached price feeds with the data in the given VAAs. This is a 
+    /// convienence wrapper around update_price_feeds(), which allows you to update the price feeds
+    /// using an entry function.
+    /// 
+    /// If possible, it is recommended to use update_price_feeds() instead, which avoids the need
+    /// to pass a signer account. update_price_feeds_with_funder() should only be used when
+    /// you need to call an entry function.
+    /// 
+    /// WARNING: this function will charge an update fee, transferring some AptosCoin's
```

Comment:
nitpick: "WARNING" seems a little scary. I think omitting that and simply saying "This function will charge..." would be better